doc/source/rllib/rllib-learner.rst

```diff
@@ -0,0 +1,459 @@
+                            )
+
+            learner_group = LearnerGroup(learner_spec)
+
+    .. tab-item:: Constructing a Learner
+
+        .. testcode::
+            :skipif: True
+
+            env = gym.make("CartPole-v1")
+
+            module_spec = SingleAgentRLModuleSpec(
+                            module_class=PPOTorchRLModule,
+                            observation_space=env.observation_space,
+                            action_space=env.action_space,
+                            model_config_dict={},
+                            catalog_class=PPOCatalog
+                        )
+
+            hparams = PPOLearnerHyperparameters(
+                use_kl_loss=True, 
+                kl_coeff=0.01,
+                kl_target=0.05, 
+                clip_param=0.2, 
+                vf_clip_param=0.2, 
+                entropy_coeff=0.05,
+                vf_loss_coeff=0.5
+            )
+
+            learner = PPOTorchLearner(
+                module_spec=module_spec, 
+                learner_hyperparameters=hparams,
+                framework_hyperparameters=FrameworkHyperparameters()
+            )
+
+Updates
+-------
+
+.. testcode::
+    :hide:
+    :skipif: True
+
+    from ray.rllib.policy.sample_batch import (DEFAULT_POLICY_ID, SampleBatch, 
+        MultiAgentBatch)
+    from ray.rllib.evaluation.postprocessing import Postprocessing
+
+    DUMMY_BATCH = {
+        SampleBatch.OBS: np.array(
+            [[0.1, 0.2, 0.3, 0.4], [0.5, 0.6, 0.7, 0.8], [0.9, 1.0, 1.1, 1.2]],
+            dtype=np.float32,
+        ),
+        SampleBatch.NEXT_OBS: np.array(
+            [[0.1, 0.2, 0.3, 0.4], [0.5, 0.6, 0.7, 0.8], [0.9, 1.0, 1.1, 1.2]],
+            dtype=np.float32,
+        ),
+        SampleBatch.ACTIONS: np.array([0, 1, 1]),
+        SampleBatch.PREV_ACTIONS: np.array([0, 1, 1]),
+        SampleBatch.REWARDS: np.array([1.0, -1.0, 0.5], dtype=np.float32),
+        SampleBatch.PREV_REWARDS: np.array([1.0, -1.0, 0.5], dtype=np.float32),
+        SampleBatch.TERMINATEDS: np.array([False, False, True]),
+        SampleBatch.TRUNCATEDS: np.array([False, False, False]),
+        SampleBatch.VF_PREDS: np.array([0.5, 0.6, 0.7], dtype=np.float32),
+        SampleBatch.ACTION_DIST_INPUTS: np.array(
+            [[-2.0, 0.5], [-3.0, -0.3], [-0.1, 2.5]], dtype=np.float32
+        ),
+        SampleBatch.ACTION_LOGP: np.array([-0.5, -0.1, -0.2], dtype=np.float32),
+        SampleBatch.EPS_ID: np.array([0, 0, 0]),
+        SampleBatch.AGENT_INDEX: np.array([0, 0, 0]),
+        Postprocessing.ADVANTAGES: np.array([0.1, 0.2, 0.3], dtype=np.float32),
+        Postprocessing.VALUE_TARGETS: np.array([0.5, 0.6, 0.7], dtype=np.float32),
+    }
+    default_batch = SampleBatch(DUMMY_BATCH)
+    DUMMY_BATCH = default_batch.as_multi_agent()
+    ADDITIONAL_UPDATE_KWARGS = {"timestep": 0, "sampled_kl_values": {DEFAULT_POLICY_ID: 1e-4}}
+
+    learner.build() # needs to be called on the learner before calling any functions
```

Comment:
Hiding this makes the example not self-contained. Users can't copy/paste it and play with it.
They'd run into the task of construction such a dummy batch themselves which is very painful. Can we make this examples complete? Right now we are just showcasing the API but I fear that it hinders accessibility and learning.

Reply:
I think sometime later I'll take all of this and move it into a file, but for the meantime, the point of this was so that I could make the code testable.